x/application/keeper/msg_server_transfer_application_stake.go

```diff
@@ -0,0 +1,59 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pokt-network/poktroll/telemetry"
+	"github.com/pokt-network/poktroll/x/application/types"
+)
+
+func (k msgServer) TransferApplicationStake(ctx context.Context, msg *types.MsgTransferApplicationStake) (*types.MsgTransferApplicationStakeResponse, error) {
+	isSuccessful := false
+	defer telemetry.EventSuccessCounter(
+		"transfer_application_stake",
+		telemetry.DefaultCounterFn,
+		func() bool { return isSuccessful },
+	)
+
+	logger := k.Logger().With("method", "TransferApplicationStake")
+
+	if err := msg.ValidateBasic(); err != nil {
+		return nil, err
+	}
+
+	_, isBeneficiaryFound := k.GetApplication(ctx, msg.Beneficiary)
+	if isBeneficiaryFound {
+		return nil, types.ErrAppDuplicateAddress.Wrapf("beneficiary (%q) exists", msg.Beneficiary)
+	}
+
+	foundApp, isAppFound := k.GetApplication(ctx, msg.Address)
+	if !isAppFound {
+		return nil, types.ErrAppNotFound.Wrapf("application %q not found", msg.Address)
+	}
+
+	beneficiary := k.createApplication(ctx, &types.MsgStakeApplication{
+		Address:  msg.Beneficiary,
+		Stake:    foundApp.Stake,
+		Services: foundApp.ServiceConfigs,
+	})
+
+	// TODO_TEST: add E2E coverage to assert #DelegateeGatewayAddresses and #PendingUndelegations
+	// are present on the beneficiary application.
+	beneficiary.DelegateeGatewayAddresses = foundApp.DelegateeGatewayAddresses
```

Comment:
Are you suggesting to support re-delegation at the same time as app stake transfer? This would have to be reconciled with app undelegation & pruning logic.

Reply:
In `MsgStakeApplication`,

```diff
message MsgStakeApplication {
  option (cosmos.msg.v1.signer) = "address";

  string address = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"]; // The Bech32 address of the application.
  cosmos.base.v1beta1.Coin stake = 2; // The total amount of uPOKT the application has staked. Must be ≥ to the current amount that the application has staked (if any)
  repeated poktroll.shared.ApplicationServiceConfig services = 3; // The list of services this application is staked to request service for
+  // TODO_POST_MAINNET_CONSIDERATION: Consdier allowing appplications to delegate
+  // to gateways at time of staking for a better developer experience. 
+  // repeated string gateway_addresss
}
```